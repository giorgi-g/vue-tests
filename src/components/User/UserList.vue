<template>
  <div>
    <ul>
      <UserListItem :key="index" v-for="(user, index) in users" :user="user" />
    </ul>
  </div>
</template>

<script>
import UserListItem from "./UserListItem";

export default {
  name: "UserList",
  components: {
    UserListItem
  },

  data() {
    return {
      users: [
        {
          name: "Tony",
          surname: "Stark",
          age: 48
        },
        {
          name: "Steve",
          surname: "Rogers",
          age: 103
        },
        {
          name: "Bucky",
          surname: "Barnes",
          age: 106
        }
      ]
    };
  }
};
</script>

<style scoped></style>
