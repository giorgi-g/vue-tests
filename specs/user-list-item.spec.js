import UserList from "@/components/User/UserList";
import { mount } from "@vue/test-utils";

test("User List Component Shallow Mount", () => {
  // As shallow and hollow as it sounds
  // console.log(shallowMount(UserList).html());

  const wrapper = mount(UserList);
  const dataUsers = wrapper.vm.$data.users;
  const user = {
    name: "G",
    surname: "Root",
    age: 200
  };

  const users = [...dataUsers, user];

  wrapper
    .setData({
      users
    })
    .then(() => {
      console.log("wrapper html", wrapper.html());

      expect(wrapper.html()).toMatchSnapshot();
    });
});
