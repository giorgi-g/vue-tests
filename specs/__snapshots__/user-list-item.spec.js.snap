// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User List Component Shallow Mount 1`] = `
<div>
  <ul>
    <li><strong>Tony Stark - 48 year(s) old</strong></li>
    <li><strong>Steve Rogers - 103 year(s) old</strong></li>
    <li><strong>Bucky Barnes - 106 year(s) old</strong></li>
    <li><strong>G Root - 200 year(s) old</strong></li>
  </ul>
</div>
`;
